From 6fb29acc8efe45a38e3a094390bf03e05b456940 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 18 May 2019 11:31:41 -0400
Subject: [PATCH 01/25] genirq: Prevent use-after-free and work list corruption
 on rt

Since commit [70a44a0 genirq: Prevent use-after-free and work list
corruption] applied, the patch makes it work on rt

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 kernel/irq/manage.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index f72bd3a357da..143f87bbff48 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -385,7 +385,11 @@ irq_set_affinity_notifier(unsigned int irq, struct irq_affinity_notify *notify)
 	raw_spin_unlock_irqrestore(&desc->lock, flags);
 
 	if (old_notify) {
+#ifdef CONFIG_PREEMPT_RT_BASE
+		kthread_cancel_work_sync(&old_notify->work);
+#else
 		cancel_work_sync(&old_notify->work);
+#endif
 		kref_put(&old_notify->kref, old_notify->release);
 	}
 
-- 
2.19.1

